apps/prairielearn/src/question-servers/freeform.js

```diff
@@ -449,7 +449,10 @@ function checkData(data, origData, phase) {
     }
     if (!editPhases.includes(phase)) {
       if (!_.has(origData, prop)) return '"' + prop + '" is missing from "origData"';
-      if (!_.isEqual(data[prop], origData[prop])) {
+
+      // If `data` and `origData` are the same object, we can skip the expensive
+      // deep comparison of them. We rely on this optimization in `processQuestionHtml`.
+      if (data[prop] !== origData[prop] && !_.isEqual(data[prop], origData[prop])) {
```

Comment:
It turns out that lodash already uses this optimization:

https://github.com/lodash/lodash/blob/f299b52f39486275a9e6483b60a410e06520c538/lodash.js#L3310

Which of course makes sense. I'll back this out.

Reply:
Removed in fccdfa5.